lib/src/robot_interface/robot_interface.cpp

```diff
@@ -327,11 +336,41 @@ void RobotInterface::IRCb(const sensor_msgs::RangeConstPtr& msg)
 
 void RobotInterface::filterForces()
 {
-    _filt_force[0] = (1 - FORCE_ALPHA) * _filt_force[0] + FORCE_ALPHA * _curr_wrench.force.x;
-    _filt_force[1] = (1 - FORCE_ALPHA) * _filt_force[1] + FORCE_ALPHA * _curr_wrench.force.y;
-    _filt_force[2] = (1 - FORCE_ALPHA) * _filt_force[2] + FORCE_ALPHA * _curr_wrench.force.z;
+    double delta_time = ros::Time::now().toSec() - _filt_updated;
+
+    std::vector<double> new_filt;
+    std::vector<double> predicted_filt;
+
+    // initial attempt to update filter using an EMA
+    new_filt.push_back((1 - FORCE_ALPHA) * _filt_force[0] + FORCE_ALPHA * _curr_wrench.force.x);
+    new_filt.push_back((1 - FORCE_ALPHA) * _filt_force[1] + FORCE_ALPHA * _curr_wrench.force.y);
+    new_filt.push_back((1 - FORCE_ALPHA) * _filt_force[2] + FORCE_ALPHA * _curr_wrench.force.z);
+
+    // use the previous rate of change of filt to predict a new filt
+    // if the prediction is 0, this is most likely the first trial so set the filt to the initial attempt
+    // otherwise, compare the prediction to the initial attempt
+    // if the attempt is within a threshold defined in utils.h, set the filt to the initial attempt
+    // this prevents the filter from changing wildly and maintains trends in the data
+
```

Comment:
These comments are a good start toward understanding what's going on, but they are not sufficient. It is not exactly clear to me what the SW is doing in the `for` loop below. Try to use comments as "Explaining what you're doing to a 15 years old person" :smile: